packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -117,12 +122,31 @@ const validTransitions = {
  * }>} MethodContext
  */
 
+/**
+ * Ephemera are the elements of state that cannot (or need not) be durable.
+ * When there's a single instance it can be held in a closure, but there are
+ * many vault manaager objects. So we hold their ephemera keyed by the durable
+ * vault manager object reference.
+ *
+ * @type {(key: VaultId) => {
+ * storageNode: ERef<StorageNode>,
+ * marshaller: ERef<Marshaller>,
+ * }} */
+// @ts-expect-error not yet defined
+const provideEphemera = makeEphemeraProvider(() => ({}));
```

Comment:
ugh. looks like a form of global mutable state. But I gather it's the least evil of the options we know of.

Reply:
yeah, the recommendation is userland until https://github.com/Agoric/agoric-sdk/issues/5759 which is low priority.

The `provideEphemera` fn is global state but not mutable. And the objects it provides are mutable, but not global. They're keyed to each virtual object. That requires some discipline but I think we have it. :)